pravega-cli/src/test/java/io/pravega/tools/pravegacli/SecureControllerCommandsTest.java

```diff
@@ -0,0 +1,170 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.tools.pravegacli;
+
+import io.pravega.client.ClientConfig;
+import io.pravega.client.admin.StreamManager;
+import io.pravega.client.stream.ScalingPolicy;
+import io.pravega.client.stream.StreamConfiguration;
+import io.pravega.client.stream.impl.DefaultCredentials;
+import io.pravega.local.LocalPravegaEmulator;
+import io.pravega.test.common.SecurityConfigDefaults;
+import io.pravega.tools.pravegacli.commands.AdminCommandState;
+import lombok.Cleanup;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.net.URI;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+@Slf4j
+public class SecureControllerCommandsTest {
+    // Security related flags and instantiate local pravega server.
+    private static final boolean restEnabled = true;
+    private static final boolean authEnabled = true;
+    private static final boolean tlsEnabled = true;
+    private static final Integer restServerPort = 9091;
+    private static LocalPravegaEmulator localPravega;
+    private static final AtomicReference<AdminCommandState> STATE = new AtomicReference<>();
+
+    @Before
+    public void setUp() throws Exception {
+        // Create the secure pravega server to test commands against.
+        LocalPravegaEmulator.LocalPravegaEmulatorBuilder emulatorBuilder = LocalPravegaEmulator.builder()
+                .controllerPort(io.pravega.test.common.TestUtils.getAvailableListenPort())
+                .segmentStorePort(io.pravega.test.common.TestUtils.getAvailableListenPort())
+                .zkPort(io.pravega.test.common.TestUtils.getAvailableListenPort())
+                .restServerPort(io.pravega.test.common.TestUtils.getAvailableListenPort())
+                .enableRestServer(restEnabled)
+                .enableAuth(authEnabled)
+                .enableTls(tlsEnabled)
+                .restServerPort(restServerPort);
+
+        // Since the server is being built right here, avoiding delegating these conditions to subclasses via factory
+        // methods. This is so that it is easy to see the difference in server configs all in one place. This is also
+        // unlike the ClientConfig preparation which is being delegated to factory methods to make their preparation
+        // explicit in the respective test classes.
+
+        if (authEnabled) {
+            emulatorBuilder.passwdFile("src/test/resources/passwd")
```

Comment:
Curious, does this path resolve properly under both these circumstances:
1. When you run this test interactively through an IDE.
2. When you run it using Gradlew command line. 

Asking since you have a relative path starting from the current directory, which may vary based on how this test is run.

Reply:
The tests run successfully both interactively through the IDE and using the gradlew command line. Is there a better way in which the same can be achieved?

Reply:
Sounds good.